api/pkg/apps/webapp/templates/form_element.gohtml

```diff
@@ -60,5 +60,261 @@
+                       type="hidden"
+                       placeholder="{{.Attributes.Placeholder}}"
+                       {{if .Validation.Required}}required{{end}}>
+
+            </div>
+        </div>
+
+        <script>
+          // since this is a component and it can be used multiple times on the same page, collisions are possible
+          // adding the code to an anonymous function such as the event listener below effectively creates a new
+          // namespace, provided variables and functions are declared using the const keyword.
+          document.addEventListener('DOMContentLoaded', (event) => {
+            const taxonomy = {
+              "ICLA": {
+                "Counselling": {
+                  "Legal Identity": {},
+                  "Refugee Status, Migration, and Residency": {},
+                  "Housing, Land and Property": {},
+                  "Employment Laws and Procedures": {},
+                  "IDP Registration": {}
+                },
+                "Information": {
+                  "Legal Identity": {},
+                  "Refugee Status, Migration, and Residency": {},
+                  "Housing, Land and Property": {},
+                  "Employment Laws and Procedures": {},
+                  "IDP Registration": {},
+                  "Essential Services": {}
+                },
+                "Legal Assistance": {
+                  "Legal Identity": {},
+                  "Refugee Status, Migration, and Residency": {},
+                  "Housing, Land and Property": {},
+                  "Employment Laws and Procedures": {},
+                  "IDP Registration": {},
+                  "Essential Services": {}
+                },
+                "Training": {
+                  "Legal Identity": {},
+                  "Refugee Status, Migration, and Residency": {},
+                  "Housing, Land and Property": {},
+                  "Employment Laws and Procedures": {},
+                  "IDP Registration": {},
+                  "Essential Services": {},
+                  "Other": {}
+                }
+              },
+              "Protection": {
+                "Individual Targeted Protection": {
+                  "Individual Protection Assistance": {},
+                  "Protection Case Management": {}
+                },
+                "Community Support": {
+                  "Community Protection Assistance": {}
+                }
+              }
+            }
+
+            // remove all the options from a select element
+            const clearSelectOptions = (selectId) => {
+              document.getElementById(selectId).innerHTML = "";
+            }
+
+            // create a new select option element
+            const createSelectOption = (value, label) => {
+              const option = document.createElement("option");
+              option.value = value;
+              option.text = label;
+              return option
```

Comment:
There are missing `;` after return statements in this file.  
For long .js snipets like this can we put it in a sepperate file?  something like this >>
`<script src="./js/myScript.js" type="text/javascript" charset="utf-8"></script> `

Reply:
I agree, we should start putting scripts like this one in their own folder 👍🏻

Reply:
That makes sense! I will implement this

Reply:
The only downside is that we can then not use golang templatign directives in the JS code.

Reply:
Good point. So instead of importing a .js script, why not template the entire `<script>` block?